homework/console/02/control.ts

```diff
@@ -0,0 +1,108 @@
+interface DoingInterface {
+  category: string;
+  content: string[];
+}
+interface MemoryInterface {
+  name: string;
+  gender: string;
+  age: number;
+  isStudent: boolean;
+  doing?: DoingInterface[];
+  hobby?: string[];
+}
+type ConvertedGender = { [key: string]: string };
```

Comment:
Mapped type 을 활용해 보고 싶었으나.. 오류가 나는데 해결 하지 못했습니다. ㅜㅜ

Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'ConvertedGender'./No index signature with a parameter of type 'string' was found on type 'ConvertedGender'.ts(7053)

```
 type GenderType = 'female' | 'male';
 type ConvertedGender = { [K in GenderType]: string };
 const convertGender: ConvertedGender = {
   female: '여성',
   male: '남성',
 };
```

Reply:
> Mapped type 을 활용해 보고 싶었으나.. 오류가 나는데 해결 하지 못했습니다. ㅜㅜ
> 
> Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'ConvertedGender'./No index signature with a parameter of type 'string' was found on type 'ConvertedGender'.ts(7053)
> 
> ```
>  type GenderType = 'female' | 'male';
>  type ConvertedGender = { [K in GenderType]: string };
>  const convertGender: ConvertedGender = {
>    female: '여성',
>    male: '남성',
>  };
> ```

작성하신코드가 오류가 없는것 같은데 이렇게 변환해서 쓰는것도 정말 좋네용 !

Reply:
앗 민오! 리뷰와주셔서 감사합니다~~

Reply:
해당 맵드 타입에서 `?:` 이렇게 붙여주면 에러는 나지 않을 거에요 ㅎㅎ 단일 타입형에선 권장되는 방법이 아닙니다. 이유는 `[key in ~~]`는 객체 형태의 타입/인터페이스의 프로퍼티 키를 찾는 구문이기 때문에 그래요!

Reply:
type ConvertedGender = { [K in GenderType]?: string }; 

이렇게 맞나용..? 에러가 안사라지는듯..싶습니닿.. 총체적 난국인가.. 🥲